### PR TITLE
AMQ-7402: Inconsistent synchronization of getters/setters of some cla…

### PR DESCRIPTION
**Pull request by PascalSchumacher**
…sses

### CODE REVIEW

**Comment by chughpiyush** [comment]
@jbonofre Is anyone working on failing test cases on Jenkins?

**Comment by jbonofre** [comment]
@chughpiyush yeah, I fixed some of then but not all, working on it (at least disabling some for now).

File: activemq-client/src/main/java/org/apache/activemq/management/SizeStatisticImpl.java [hunk 1]
```diff
@@ -87,7 +87,7 @@ public synchronized void setTotalSize(long size) {
     /**
      * @return the maximum size of any step
      */
-    public long getMaxSize() {
+    public synchronized long getMaxSize() {
```
Comment (coheigea) [inline activemq-client/src/main/java/org/apache/activemq/management/SizeStatisticImpl.java]: For Max + Min Size, I think synchronized should be removed from the setter methods and just make the variables volatile instead.

File: activemq-kahadb-store/src/main/java/org/apache/activemq/store/kahadb/disk/index/HashIndex.java [hunk 2]
```diff
@@ -383,7 +383,7 @@ int indexFor(Key x, int length) {
     // Property Accessors
     // /////////////////////////////////////////////////////////////////
 
-    public Marshaller<Key> getKeyMarshaller() {
+    public synchronized Marshaller<Key> getKeyMarshaller() {
```
Comment (coheigea) [inline activemq-kahadb-store/src/main/java/org/apache/activemq/store/kahadb/disk/index/HashIndex.java]: Instead remove synchronized from the setter, and make the variable volatile

**Pull request merged**